### PR TITLE
[LA][APB-500] Implementation and integration tests for subscription endpoint

### PR DESCRIPTION
**Pull request by bravesirrobin81**

### CODE REVIEW

File: app/uk/gov/hmrc/agentsubscription/model/SubscriptionRequest.scala [hunk 1]
```diff
@@ -39,8 +39,7 @@ case class Address(addressLine1: String, addressLine2: String, addressLine3: Opt
 case class SubscriptionRequest(name: String,
                          address: Address,
```
Comment (bravesirrobin81) [inline app/uk/gov/hmrc/agentsubscription/model/SubscriptionRequest.scala]: I'm in two minds about whether to keep the domain models for this service and for DES separate. Opinions welcome!

**Comment by frankoid** [comment]
The API implemented here doesn't seem to match the API described in README.md.

The API implemented looks quite like the DES one, but I think that we should make the agent-subscription API look like we want it to rather than matching the DES one. This may be relevant to your question about whether the keep the domain models for the service separate - I think we probably should (unless the DES one is pretty much what we would've come up with anyway).

**Comment by frankoid** [comment]
also I think that the subscription endpoint should be checking known facts (postcode) as a security precaution

**Comment by bravesirrobin81** [comment]
That's more down to the fact that the Readme hadn't been updated. I wasn't aware that there was documentation of the API in there as well as the RAML. 

I'm actually starting to question the value of having this service at all. If we're doing the same set of checks here as in the frontend, and exposing the same API as DES, then it feels like twice as much work for no gain.

**Comment by frankoid** [comment]
We sketched out the API in the README a couple of weeks ago, but you weren't to know as you weren't around.

The main aspect of the README that I'd like to preserve is that the request body has an `agency` section containing properties of the _agency subscription_ (which the name/address/email/telephone fields in the API implemented here should probably be moved into) and a `knownFacts` section containing known facts about the _registration_ of the entity that is subscribing.

I'm not too bothered about whether the UTR is in the URL or the request body.

If the subscribe endpoint in the backend checks the known facts and then calls the DES API to subscribe then I think it's doing something of value (and not just exposing the same API as DES).

On a more mundane note, I don't think frontends can call DES.

**Pull request merged**